lib/Interpreter/CppInterOp.cpp

```diff
@@ -930,6 +986,121 @@ namespace Cpp {
     return true;
   }
 
+  std::vector<TCppFunction_t> GetTemplatedMethods(const std::string& name,
+                                                  TCppScope_t parent,
+                                                  const std::string& filter) {
+
+    auto* D = (Decl*)parent;
+
+    if (!parent || name.empty())
+      return {};
+
+    D = GetUnderlyingScope(D);
+
+    std::vector<TCppFunction_t> funcs;
+    llvm::StringRef Name(name);
+    auto& S = getSema();
+    DeclarationName DName = &getASTContext().Idents.get(name);
+    clang::LookupResult R(S, DName, SourceLocation(), Sema::LookupOrdinaryName,
+                          Sema::ForVisibleRedeclaration);
+
+    Cpp_utils::Lookup::Named(&S, R, Decl::castToDeclContext(D));
+
+    if (R.empty())
+      return {};
+
+    R.resolveKind();
+    TCppFunction_t best_match;
+
+    for (auto* Found : R) {
+      if (llvm::isa<FunctionTemplateDecl>(Found))
+        funcs.push_back(Found);
+    }
+
+    if (!funcs.empty())
+      return funcs;
+
+    return {};
+  }
+
+  TCppIndex_t GetNumTemplatedMethods(TCppScope_t scope, bool accept_namespace) {
+    if (!accept_namespace && IsNamespace(scope))
+      return (TCppIndex_t)0; // Enforce lazy
+
+    std::vector<TCppFunction_t> mc = GetTemplatedFuncs(scope);
+    TCppIndex_t res = 0;
+
+    for (auto method : mc) {
+      res += IsTemplatedFunction(method) ? 1 : 0;
```

Comment:
Can `mc` contain non-templated functions?

Reply:
It should contain only templated functions, usecase is like so:

```cpp
// add proxies for un-instantiated/non-overloaded templated methods
    const Cppyy::TCppIndex_t nTemplMethods = isNamespace ? 0 : Cppyy::GetNumTemplatedMethods(scope);
 ```
 There is an existing cling implementation here :
 https://github.com/wlav/cppyy-backend/blob/4656c034c05f99a1040a2a506e2cdf5a2ebbae8d/clingwrapper/src/clingwrapper.cxx#L1924

Reply:
Then why this check if that's a templated function when we know it is?

Reply:
Ah I missed that, yes mc is obtained using GetFunctionTemplatedDecls. We do not need this check here